### DISCUSSION TITLE
hidden field in admin ui - table row is still rendered

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#4990] [Q&A Core]
Hi,
I have hidden field in Admin UI defined this way (4 fields at all):

```
		'widget_position'		=> array(
			'tab' => 0, 'data' => false ,   
		 
		),
```

![image](https://user-images.githubusercontent.com/5429548/228059104-519e9b7e-058f-4187-84e5-1f35b44ecfc2.png)

but row is still rendered 

![image](https://user-images.githubusercontent.com/5429548/228059247-ee58f7f1-9b30-438d-85c7-3394ebc868b2.png)

output for one hidden field is:
```
<tr>
					<td>
						<span></span>
					</td>
					<td>
						
					</td>
				</tr>
```

Am I missing something in field definition?

Thanks

**Comment by Moc** [comment]
Did you try `'type' => 'hidden',`?

**Comment by Jimmi08** [comment]
@Moc The point was - documentation is saying that it is hidden but it is not. At least some css class or id for tr tag to be able hide it manually. 

I can't use type hidden because it adds new input element to form (only type hidden, so you don't see it).  This input field is added in other method field (it is in fact select type field) so with hidden type - the name would be there 2x. 

Yeah, it is advanced and I already asked for example how to do form when one field depends on inserted value of other field (subforum field is nice example, issue is still open), so I did it other way. And everything works nicely except those empty an reductant html tags for not displayed field. 

For you:
this is example from method field:

```
		$text = '<tr valign="top" cli_frm_tgl-id="cli_bar_type" cli_frm_tgl-val="widget" style="display: none;">
						<th scope="row"><label for="widget_position">'. LAN_GC_MESSAGEBAR_012. '</label></th>
						<td>
							<select name="widget_position" id="widget_position" class="vvv_combobox">
								<option value="left" ' . $types['left'] .  '>'. LAN_GC_MESSAGEBAR_019.'</option>
								<option value="right" ' . $types['right'].  '>'. LAN_GC_MESSAGEBAR_020.'</option>
							</select>
						</td>
				</tr>';
		return $text;
```

I just got idea - to count tr tags and hide nth tr tag of admin UI table.  When I am back to this, I will try it.

**Reply by Vodhin** [reply]
I use special classes and jQuery to hide any matched TR (and also add toggles to show or hide if needed) eg:
`'writeParms' =>  array ('size' => 'xxlarge specialClass1')`
and then 
```
$(document).ready(function(){
  $('.specialClass1').closest('tr').hide();
  })
```

**Reply by Jimmi08** [reply]
Yes, I am familiar with similar solution. 
 
For me, this is bug, I am just waiting for confirmation of core developer.
